conans/test/integration/build_requires/build_requires_test.py

```diff
@@ -451,3 +451,55 @@ def build(self):
                     .with_build_requires("harfbuzz/1.0@test/test")})
         client.run("install . --build=missing")
         self.assertIn("ZLIBS LIBS: ['myzlib']", client.out)
+
+
+def test_dependents_new_buildenv():
+    client = TestClient()
+    boost = textwrap.dedent("""
+        from conans import ConanFile
+        class Boost(ConanFile):
+            def package_info(self):
+                self.buildenv_info.define_path("PATH", "myboostpath")
+        """)
+    other = textwrap.dedent("""
+        from conans import ConanFile
+        class Other(ConanFile):
+            requires = "boost/1.0"
+            def package_info(self):
+                self.buildenv_info.append_path("PATH", "myotherpath")
```

Comment:
Might be worthy to test what happens with prepending

Reply:
Added prepend testing (in the same test, to see how it behaves together with other append)